deploy/crds/keycloak.org_keycloakbackups_crd.yaml

```diff
@@ -10,144 +10,144 @@ spec:
-                    S3 Service. The secret needs to be in the following form: \n     apiVersion:
-                    v1     kind: Secret     metadata:       name: <Secret name>     type:
-                    Opaque     stringData:       AWS_S3_BUCKET_NAME: <S3 Bucket Name>
-                    \      AWS_ACCESS_KEY_ID: <AWS Access Key ID>       AWS_SECRET_ACCESS_KEY:
-                    <AWS Secret Key> \n For more information, please refer to the
-                    Operator documentation."
-                  type: string
-                encryptionKeySecretName:
-                  description: "If provided, the database backup will be encrypted.
-                    Provides a secret name used for encrypting database data. The
-                    secret needs to be in the following form: \n     apiVersion: v1
-                    \    kind: Secret     metadata:       name: <Secret name>     type:
-                    Opaque     stringData:       GPG_PUBLIC_KEY: <GPG Public Key>
-                    \      GPG_TRUST_MODEL: <GPG Trust Model>       GPG_RECIPIENT:
-                    <GPG Recipient> \n For more information, please refer to the Operator
-                    documentation."
-                  type: string
-                schedule:
-                  description: If specified, it will be used as a schedule for creating
-                    a CronJob.
-                  type: string
-              type: object
-            instanceSelector:
-              description: Selector for looking up Keycloak Custom Resources.
-              properties:
-                matchExpressions:
-                  description: matchExpressions is a list of label selector requirements.
-                    The requirements are ANDed.
-                  items:
-                    description: A label selector requirement is a selector that contains
-                      values, a key, and an operator that relates the key and values.
-                    properties:
-                      key:
-                        description: key is the label key that the selector applies
-                          to.
-                        type: string
-                      operator:
-                        description: operator represents a key's relationship to a
-                          set of values. Valid operators are In, NotIn, Exists and
-                          DoesNotExist.
-                        type: string
-                      values:
-                        description: values is an array of string values. If the operator
-                          is In or NotIn, the values array must be non-empty. If the
-                          operator is Exists or DoesNotExist, the values array must
-                          be empty. This array is replaced during a strategic merge
-                          patch.
-                        items:
+  versions:
+  - name: v1alpha1
```

Comment:
Do you think the version should be bumped with this switch to `v1`?

The same applies to the other CRDs as well.

Reply:
This is something that has been introduced in Operator SDK 18.0.0 (see the [release notes](https://github.com/operator-framework/operator-sdk/releases/tag/v0.18.0)). This should be fine for K8s `1.16+`. Since we use `alpha` for our CRDs, I think that's a safe change. WDYT?

Reply:
@slaskawi More operators are going to be going through the upgrade of CRDs to using `v1`. I believe some fields in the spec may have been removed. 

https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1beta1/types.go vs https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types.go

Since the Keycloak CRDs are `v1alpha1`, you can make as many changes as you want. Might still be worth bumping the version depending on the diff between the above links.

Reply:
Let me ask @stianst for opinion here.

Reply:
Let me put this on hold for a moment. It just came to me that `apiVersion` change might be problematic from OLM upgrade point of view. I asked Operator SDK Team to clarify this.

Reply:
We had a chat internally about this migration. There are two important aspects:
- The API Server will automatically and transparently convert CRDs between `v1beta1` and `v1`. There should be no issues from this perspective.
- OpenShift 4.3 and 4.4 (that we use for the testing in the product) requires `v1beta1`. This means it's our lower bound and we simply can not migrate to `v1` at this stage.

@davidffrench @pb82 This might be also important to you. You will probably have a similar conversation about upgrades soon.

//cc @abstractj @vmuzikar @ASzc

Reply:
Perfect, thank you @slaskawi , this is fantastic information. I was following your thread on the operator framework mailing list as well.